plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergMaterializedViewTest.java

```diff
@@ -737,6 +737,43 @@ public void testMaterializedViewSnapshotSummariesHaveTrinoQueryId()
         }
     }
 
+    @Test
+    public void testMaterializedViewStorageTypeCoercions()
```

Comment:
this is really engine fix. Could you add test to `TestAnalyzer` or `TestMaterializedViews` as in https://github.com/trinodb/trino/pull/8408

Reply:
Added test to TestMaterializedViews